File: Museum/Frontend/museum-frontend-angular/src/app/models/WeatherDTO.ts
interface Weather{
    id: number;
    main: string;
    description: string;
    icon: string;
}

interface Main{
    temp: number;
    feels_like: number;
    temp_min: number;
    temp_max: number;
    pressure: number;
    humidity: number;
}


export interface WeatherDTO{
    weather: Weather[];
    main: Main;
    wind: {
        speed: number;
    }
    name: string;
}

File: Museum/Backend/REST service/src/main/java/museum/service/models/DTOs/WeatherDTO.java
package museum.service.models.DTOs;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
@NoArgsConstructor
public class WeatherDTO
{
    @Data
    @NoArgsConstructor
    public static class Weather
    {
        private Integer id;
        private String main;
        private String description;
        private String icon;
    }
    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)
    @NoArgsConstructor
    public static class Main
    {
        private Double temp;
        private Double feels_like;
        private Double temp_min;
        private Double temp_max;
        private Double pressure;
        private Double humidity;
    }
    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)
    @NoArgsConstructor
    public static class Wind
    {
        private Double speed;
    }

    private List<Weather> weather;
    private Main main;
    private Wind wind;
    private String name;
}
